examples/quickstart/src/main/java/com/okta/jwt/example/QuickStartExample.java

```diff
@@ -50,5 +54,24 @@ public static void main(String[] args) throws Exception {
         System.out.println(jwt.getClaims().get("invalidKey")); // an invalid key just returns null
         System.out.println(jwt.getClaims().get("groups")); // handle an array value
         System.out.println(jwt.getExpiresAt()); // print the expiration time
+        
+        
+        // To decode IdToken - nonce is required. It will be available as a cookie if you have used Okta SignIn Widget.
+        // For others, please check the nonce value sent in request.
```

Comment:
While correct this _might_ be a little miss leading.  For that use case, you should likely be using access tokens instead of ID tokens. (Usually you do not want to pass around Id-tokens).

This is still a good PR though, when I’m back at my desk I’ll try to provide some more concrete  info.

Reply:
Thank you. I do have one usecase to validate the id token. Id tokens are basically used to gather user information. Until, I validate and decode it, I cannot get the user info with current library implementation.

I can simply use the Jwt parsers offered by other libraries and get the information but I will still have to ensure the authenticity of the token as per the standards.

May be, the comment could be misleading but that's my observation on where to get the nonce value from. If there is a pre-defined way to pass this or get this value from, I would be glad to know that.

Reply:
Can you tell me about your choice to use an ID token vs an access token?  (you can include additional information in an access token too if you are trying to reduce requests back to the IdP i.e. Okta.

You shouldn't depend on getting the `nonce` from a cookie as that is an implementation detail and could change, along with this a `nonce` _should_ only be used once. (and not passed between applications)

In general, using this lib to validate ID Tokens should only happen if you are building an OpenID connect client and want to validate, i.e. calling the [`/token` endpoint](https://developer.okta.com/docs/reference/api/oidc/#token). 

If you are building a SPA that contains a custom login page (e.g. using the Okta Sign-In Widget) you should be using the access token (typically in a header `Authorization: Bearer <your-access-token>`)

I'm making some assumptions here, so let's take a step back and tell me about your ID-Token use case first.

Reply:
we may pass only the access token to an api where user identity does not come into picture but what if the API has to know the user info/profile and based on that it performs some business logic.

Do we have to check with the auth server to get the user profile?

Reply:
You have a few options:

1. Include that info in your access tokens ([for more info, take a look at the docs](https://developer.okta.com/docs/guides/customize-authz-server/create-claims/)]

2. Make a request to the [/introspect](https://developer.okta.com/docs/reference/api/oidc/#introspect) or [/userinfo](https://developer.okta.com/docs/reference/api/oidc/#alternative-validation) endpoints

3. Initiate the user login from the server side (your application or something like an API gateway)

If you choose option 2 or 3 using an OIDC/OAuth 2.0 library might be a better fit for your application.


Again, I'm making some assumptions here 😉  so please let me know if I can answer any questions.  Also, let me know which Java frameworks you are building on and we might be able to give you more specific pointers. 😄

Reply:
May be it is not as per the standards. Thanks for the information.